homeassistant/components/tedee/config_flow.py

```diff
@@ -0,0 +1,309 @@
+"""Config flow for Tedee integration."""
+from collections.abc import Mapping
+from typing import Any
+
+from pytedee_async import TedeeAuthException, TedeeClient, TedeeLocalAuthException
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_HOME_ASSISTANT_ACCESS_TOKEN,
+    CONF_LOCAL_ACCESS_TOKEN,
+    CONF_UNLOCK_PULLS_LATCH,
+    CONF_USE_CLOUD,
+    DOMAIN,
+    NAME,
+)
+
+
+async def validate_input(user_input: dict[str, Any]) -> bool:
+    """Validate the user input allows us to connect."""
+    pak = user_input.get(CONF_ACCESS_TOKEN, "")
+    host = user_input.get(CONF_HOST, "")
+    local_access_token = user_input.get(CONF_LOCAL_ACCESS_TOKEN, "")
+    tedee_client = TedeeClient(pak, local_access_token, host)
+
+    try:
+        await tedee_client.get_locks()
+    except (TedeeAuthException, TedeeLocalAuthException) as ex:
+        raise InvalidAuth from ex
+    except Exception as ex:
+        raise CannotConnect from ex
+    return True
+
+
+class TedeeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tedee."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._errors: dict = {}
+        self._reload: bool = False
+        self._previous_step_data: dict = {}
+        self._config: dict = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            if (
+                user_input.get(CONF_HOST) is None
+                and user_input.get(CONF_LOCAL_ACCESS_TOKEN) is not None
+            ):
+                errors[CONF_HOST] = "invalid_host"
+            elif (
+                user_input.get(CONF_HOST) is not None
+                and user_input.get(CONF_LOCAL_ACCESS_TOKEN) is None
+            ):
+                errors[CONF_LOCAL_ACCESS_TOKEN] = "invalid_api_key"
+            elif (
+                user_input.get(CONF_HOST) is not None
+                and user_input.get(CONF_LOCAL_ACCESS_TOKEN) is not None
+            ):
+                try:
+                    await validate_input(user_input)
+                except InvalidAuth:
+                    errors[CONF_LOCAL_ACCESS_TOKEN] = "invalid_api_key"
+                except CannotConnect:
+                    errors[CONF_HOST] = "invalid_host"
```

Comment:
```suggestion
            try:
                if user_input.get(CONF_HOST) is None:
                    raise CannotConnect
                if user_input.get(CONF_LOCAL_ACCESS_TOKEN) is None:
                    raise InvalidAuth
                
                await validate_input(user_input)
            except InvalidAuth:
                errors[CONF_LOCAL_ACCESS_TOKEN] = "invalid_api_key"
            except CannotConnect:
                errors[CONF_HOST] = "invalid_host"
```

Not sure if this is _exactly_ the same, but IMO it is more readable if it does

Reply:
not the same. My code ensures that if one of `CONF_HOST`, `CONF_LOCAL_ACCESS_TOKEN` is set, the other must be set as well, but if you leave both of them empty that's fine (you must set the `INIT_CLOUD` then though). Your code would make any of them mandatory.